block/manager_test.go

```diff
@@ -94,3 +98,94 @@ func TestIsDAIncluded(t *testing.T) {
 	m.blockCache.setDAIncluded(hash.String())
 	require.True(m.IsDAIncluded(hash))
 }
+
+func TestSubmitBlocks(t *testing.T) {
+	require := require.New(t)
+	ctx := context.Background()
+	logger := test.NewFileLoggerCustom(t, test.TempLogFileName(t, t.Name()))
+
+	// Create a minimalistic block manager
+	m := &Manager{
+		dalc:       &da.DAClient{DA: goDATest.NewDummyDA(), GasPrice: -1, Logger: logger},
+		blockCache: NewBlockCache(),
+		logger:     logger,
+	}
+
+	maxDABlobSizeLimit, err := m.dalc.DA.MaxBlobSize(ctx)
+	require.NoError(err)
+
+	testCases := []struct {
+		name                string
+		blocks              []*types.Block
+		isErrExpected       bool
+		expectedNumAttempts uint64
```

Comment:
ok so looking at the tests I don't think we need this. 
it isn't necessarily important how many attempts it takes for these test cases. Just that the expect outcome is achieved. 
This means we can just remove this from `submitBlocksToDA`.

Reply:
also I think we should be checking pending blocks anyways to confirm that the pending blocks are expected. either empty of containing blocks that weren't submitted.